js/utils/oppfolgingsforlopsperioderUtils.js

```diff
@@ -0,0 +1,64 @@
+import {
+    finnArbeidsgivereForAktiveSykmeldinger,
+} from './sykmeldingUtils';
+
+export const hentOppfolgingsPerioderFeilet = (oppfolgingsforlopsPerioderReducerListe) => {
+    return oppfolgingsforlopsPerioderReducerListe.filter((reducer) => {
+        return reducer.hentingFeilet;
+    }).length > 0;
+};
+
+export const forsoektHentetOppfolgingsPerioder = (oppfolgingsforlopsPerioderReducerListe) => {
+    return oppfolgingsforlopsPerioderReducerListe.filter((reducer) => {
+        return reducer.hentet || reducer.hentingFeilet;
+    }).length === oppfolgingsforlopsPerioderReducerListe.length;
+};
+
+export const henterEllerHarForsoektHentetOppfolgingsPerioder = (oppfolgingsforlopsPerioderReducerListe) => {
+    return oppfolgingsforlopsPerioderReducerListe.filter((reducer) => {
+        return reducer.henter || forsoektHentetOppfolgingsPerioder([reducer]);
+    }).length === oppfolgingsforlopsPerioderReducerListe.length;
+};
+
+export const finnVirksomheterMedAktivSykmelding = (sykmeldinger, ledere) => {
+    return finnArbeidsgivereForAktiveSykmeldinger(sykmeldinger, ledere).map((virksomhet) => {
+        return virksomhet.virksomhetsnummer;
+    });
+};
+
+export const finnOgHentManglendeOppfolgingsforlopsPerioder = (props) => {
```

Comment:
Denne logikken kan heller gjøres i en saga der du har tilgang til hele state. Da syns jeg det blir enklere å forstå hva som skjer. Når funksjonen tar imot props som allerede er manipulert, er det vanskelig å forstå hva de forskjellige argumentene er. Hva er oppfolgingsforlopsPerioderReducerListe? Hvordan er den utledet? Hva er virksomhetsnrListe? Hvordan er denne utledet?

Reply:
Det er virksomheter man skal hente oppfolgingsforlopPerioder for, som er avhengig av både sykmeldinger og ledere.